pinot-filesystem/src/main/java/org/apache/pinot/filesystem/PinotFS.java

```diff
@@ -105,7 +105,7 @@ public abstract long length(URI fileUri)
       throws IOException;
 
   /**
-   * Lists all the files at the location provided. Lists recursively.
+   * Lists all the files and directories at the location provided. Lists recursively.
```

Comment:
Lists recursively if "recursive" is set to true.

Also, if the URI provided is not a directory, are we expected to throw exception no matter what the value of 'recursive' is? i think the localfs will not throw exception if recursive is set to false